hermes/crates/cbork/cddl-parser/src/grammar/rfc_8610.pest

```diff
@@ -103,58 +105,33 @@ int = ${ "-"? ~ uint }
 
 /// Unsigned Integers
 uint = ${
-    ( ASCII_NONZERO_DIGIT ~ ASCII_DIGIT* )
-    | ( "0x" ~ ASCII_HEX_DIGIT+ )
-    | ( "0b" ~ ASCII_BIN_DIGIT+ )
+    (ASCII_NONZERO_DIGIT ~ ASCII_DIGIT*)
+    | ("0x" ~ ASCII_HEX_DIGIT+)
+    | ("0b" ~ ASCII_BIN_DIGIT+)
     | "0"
 }
 
 /// Literal Text
 text = ${ "\"" ~ SCHAR* ~ "\"" }
 
-/// Literal Bytes - Note CDDL Spec incorrectly defines b64''.
-bytes = ${ bytes_hex | bytes_b64 | bytes_text }
-bytes_hex = ${ "h" ~ "'" ~ HEX_PAIR* ~ "'" }
-bytes_b64 = ${ "b64" ~ "'" ~ URL_BASE64 ~ "'" }
-bytes_text = ${ "'" ~ BCHAR* ~ "'" }
+/// Literal Bytes.
+bytes = ${ bsqual? ~ "'" ~ BCHAR* ~ "'" }
+bsqual = { "h" | "b64" }
 
 // -----------------------------------------------------------------------------
 // Simple multiple character sequences
 
-/// identifier,  called the `name` in the CDDL spec.
-id = ${
-    group_socket |
-    type_socket |
-    name
-}
-
-/// Special form of a name that represents a Group Socket.
-group_socket = ${ "$$" ~ ( ( "-" | "." )* ~ NAME_END )* }
-/// Special form of a name that represents a Type Socket.
-type_socket = ${ "$" ~ ( ( "-" | "." )* ~ NAME_END )* }
-/// General form of a name.
-name = ${ NAME_START ~ ( ( "-" | "." )* ~ NAME_END )* }
-
-/// A pair of hex digits. (Must always have even numbers of hex digits.)
-HEX_PAIR = _{ S ~ ASCII_HEX_DIGIT ~ S ~ ASCII_HEX_DIGIT ~ S }
-
-/// Whitespace is allowed and is ignored.
-/// This token will keep the whitespace, so it will need to handled when converted to binary.
-URL_BASE64 = _{ S ~ ( URL_BASE64_ALPHA ~ S)* ~ URL_BASE64_PAD? }
+/// identifier, called the `name` in the CDDL spec.
+id = ${ NAME_START ~ (("-" | ".")* ~ NAME_END)* }
 
 
 // -----------------------------------------------------------------------------
 // Characters, Whitespace and Comments
 
 S = _{ WHITESPACE* }
-WHITESPACE = _{ " " | "\t" | NEWLINE }
+WHITESPACE = _{ " " | "\t" | COMMENT | NEWLINE }
```

Comment:
Because from the original ABNF, the whitespace rule also contains the comment rule. That's why I made the top rules to be compound atomic all. And it has a case like:

```
a = { ; just want to comment here <----- caused parsing error
   b
}
```

Reply:
I understand.  However COMMENT can be important to retain in the AST.  If its literal whitespace it is literally stripped.
Doing this will prevent us from putting information in the comments that can be read by (for example) a code generator for producing rust code from the cddl.

Reply:
Also, note this:
https://github.com/cbor-wg/cddl-modules/blob/main/draft-ietf-cbor-cddl-modules.md
Also uses COMMENT to embed control information specifically the `;#` token.  So it may be that the easiest way to process this will be with the semantic parser, but that will ONLY work if comments are properly retained and in the proper place within the AST.